packages/builder/src/components/portal/onboarding/TourWrap.svelte

```diff
@@ -1,39 +1,62 @@
 <script>
   import { tourHandler } from "./tourHandler"
-  import { TOURS } from "./tours"
+  import { TOURSBYSTEP, TOURS, getCurrentStepIdx } from "./tours"
   import { onMount, onDestroy } from "svelte"
   import { builderStore } from "stores/builder"
 
-  export let tourStepKey
+  export let stepKeys = []
 
-  let currentTourStep
   let ready = false
-  let registered = false
-  let handler
+  let registered = {}
 
   const registerTourNode = (tourKey, stepKey) => {
-    if (ready && !registered && tourKey) {
-      currentTourStep = TOURS[tourKey].steps.find(step => step.id === stepKey)
-      if (!currentTourStep) {
-        return
-      }
-      const elem = document.querySelector(currentTourStep.query)
-      handler = tourHandler(elem, stepKey)
-      registered = true
+    const step = TOURSBYSTEP[stepKey]
+    if (ready && step && !registered[stepKey] && step?.tour === tourKey) {
+      const elem = document.querySelector(step.query)
+      registered[stepKey] = tourHandler(elem, stepKey)
     }
   }
 
   $: tourKeyWatch = $builderStore.tourKey
-  $: registerTourNode(tourKeyWatch, tourStepKey, ready)
+  $: tourStepKeyWatch = $builderStore.tourStepKey
+  $: if (tourKeyWatch || stepKeys || ready) {
+    stepKeys.forEach(tourStepKey => {
+      registerTourNode(tourKeyWatch, tourStepKey)
+    })
+  }
+
+  $: if (tourKeyWatch || tourStepKeyWatch) {
+    let tourStepIdx = getCurrentStepIdx(
+      TOURS[tourKeyWatch]?.steps,
+      tourStepKeyWatch
+    )
+    let currentStep = TOURS[tourKeyWatch]?.steps?.[tourStepIdx]
+    if (currentStep?.scrollIntoView) {
+      let currentNode = $builderStore.tourNodes?.[currentStep.id]
+      if (currentNode) {
+        currentNode.scrollIntoView({ behavior: "smooth", block: "center" })
+      }
+    }
+  }
```

Comment:
Might be better to pull this out into a function - I don't think big multi-line reactive statements are a great idea. This will re-evaluate for any change to the `builderStore` for example, which would mean that we scroll the node into view even if a completely unrelated change to the builder store is made while the other conditions are satisfied.

I think it would make sense to pull it into a separate function and call it in a reactive statement, passing in only those observed params that we care about.